WatsonWebsocket/WatsonWsClient.cs

```diff
@@ -1,472 +1,813 @@
+
+        /// <summary>
+        /// Send binary data to the server asynchronously.
+        /// </summary>
+        /// <param name="data">Byte array containing data.</param>
+        /// <param name="token">Cancellation token allowing for termination of this request.</param>
+        /// <returns>Task with Boolean indicating if the message was sent successfully.</returns>
+        public async Task<bool> SendAsync(byte[] data, CancellationToken token = default)
+        {
+            if (data == null || data.Length < 1) throw new ArgumentNullException(nameof(data));
+            return await MessageWriteAsync(data, WebSocketMessageType.Binary, token);
+        }
+
+        /// <summary>
+        /// Send text data to the server asynchronously and wait for anything response.
+        /// </summary>
+        /// <param name="data">String data.</param>
+        /// <param name="timeout">Maximum waiting time until one response.</param>
+        /// <param name="token">Cancellation token allowing for termination of this request.</param>
+        /// <returns>Task with Boolean indicating if the message was sent successfully.</returns>
+        public async Task<string> SendAndReplyAnythingAsync(string data, int timeout = 30, CancellationToken token = default)
+        {
+            if (data == null || data.Length < 1) throw new ArgumentNullException(nameof(data));
+            if (timeout < 1) throw new ArgumentException("Timeout invalid", nameof(data));
+            string result = null;
+
+            await Task.Run(async () =>
+            {
+                // Create Catcher Data
+                var receivedEvent = new ManualResetEvent(false);
+                InternalMessageReceived += (s, e) =>
+                {
+                    result = Encoding.UTF8.GetString(e.Data);
+                    receivedEvent.Set();
+                };
+                // Detect Connection Lost
+                InternalServerDisconnected += (s, e) => receivedEvent.Set();
+
+                await MessageWriteAsync(Encoding.UTF8.GetBytes(data), WebSocketMessageType.Binary, token);
+                receivedEvent.WaitOne(TimeSpan.FromSeconds(timeout));
+
+                // Reset Internal Events
+                InternalMessageReceived = delegate { };
+                InternalServerDisconnected = delegate { };
+            }, token);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Send text data to the server asynchronously and wait for text response.
+        /// </summary>
+        /// <param name="data">String data.</param>
+        /// <param name="timeout">Maximum waiting time until one response.</param>
+        /// <param name="token">Cancellation token allowing for termination of this request.</param>
+        /// <returns>Task with Boolean indicating if the message was sent successfully.</returns>
+        public async Task<string> SendAndReplyAsync(string data, int timeout = 30, CancellationToken token = default)

```

Comment:
Rather than repeat the code, ```string data``` should be converted to bytes, and then passed to the ```SendAndReplyAsync``` that accepts a byte array

Reply:
It depends on the type of message with WebSocketMessageType